intel-sgx/ias/src/verifier/mod.rs

```diff
@@ -0,0 +1,754 @@
+/* Copyright (c) Fortanix, Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::api::{EnclaveQuoteBody, IasAdvisoryId, IasVersion, PlatformStatus, QuoteStatus,
+    SUPPORTED_IAS_VERSIONS, Unverified, Verified, VerificationType, VerifyAttestationEvidenceResponse};
+use serde::Deserialize;
+use serde_bytes_repr::ByteFmtDeserializer;
+#[cfg(feature = "manipulate_attestation")]
+use std::str::FromStr;
+use std::convert::TryInto;
+use std::marker::PhantomData;
+use once_cell::sync::Lazy;
+use pkix::{ASN1Result, DerWrite, FromDer};
+use pkix::derives::{BERDecodable, BERReader, DERWriter};
+use pkix::x509::GenericCertificate;
+use sgx_isa::{Attributes, AttributesFlags, Miscselect};
+#[cfg(target_env = "sgx")]
+use sgx_isa::Report;
+use sgx_pkix::oid;
+use std::fmt;
+use std::result::Result;
+
+pub mod crypto;
+pub use self::crypto::private::Crypto;
+
+pub static MITIGATED_SECURITY_ADVISORIES: Lazy<Vec<IasAdvisoryId>> = Lazy::new(|| {
+    #[allow(unused_mut)]
+    let mut v: Vec<IasAdvisoryId> = Vec::new();
+
+    #[cfg(intel_sa_00334)]
+    v.push(IasAdvisoryId::from("INTEL-SA-00334"));
+
+    #[cfg(intel_sa_00615)]
+    v.push(IasAdvisoryId::from("INTEL-SA-00615"));
+
+    v
+});
```

Comment:
This may be a bit far-fetched, but are we sure the entire enclave is compiled with the SW mitigation in place when `#[cfg(intel_sa_00334)]`/`#[cfg(intel_sa_00615)]` are true when compiling *this specific* crate?

In other words: should the list of mitigated security advisories instead be determined by the *users* of this crate, or is there no realistic risk of this going wrong?

Reply:
Interesting comment! I tried this on a random crate:
```
cargo +nightly-2022-08-29-x86_64-unknown-linux-gnu build
# Resulted in the compilation of 75 crates
cargo +nightly-2022-08-30-x86_64-unknown-linux-gnu build
# Resulted in recompilation of 75 crates
```
So the dangers of only partially compiling the enclave with the correct toolchain is minimal. There is however always chances of linking in pre-compiled blobs with the necessary compilation options, relying on other compilers without proper mitigations for languages other then Rust whose result will be linked in (e.g., C code), ... That would be problematic for LVI mitigations (INTEL-SA-00334).
Other things that could go wrong are for example, enclaves that use unsafe code to bypass our abstractions and directly access userspace without mitigating stale data reads correctly (INTEL-SA-00615).
Hmmm I'll add a comment in the documentation for the crate. Or would you force developers to enable automatic detection of these mitigations based on the Rust compiler?

Reply:
Yes I agree that there's no concern if `AutoDetect` is removed. I assume the `MITIGATED_SECURITY_ADVISORIES` global static will then be removed as well?